packages/search/src/search.tsx

```diff
@@ -90,349 +96,349 @@ type State = {
+	const shouldRenderRightOpenIcon = openIconSide === 'right' && ! keyword;
 
-		return (
-			<div dir={ this.props.dir } className={ searchClass } role="search">
-				<Spinner />
-				{ this.props.openIconSide === 'left' && this.renderOpenIcon() }
-				<form className={ fadeClass } action="." onSubmit={ this.handleSubmit }>
-					<input
-						type="search"
-						id={ 'search-component-' + this.instanceId }
-						autoFocus={ this.props.autoFocus } // eslint-disable-line jsx-a11y/no-autofocus
-						aria-describedby={ this.props.describedBy }
-						aria-label={ inputLabel ? inputLabel : this.props.__( 'Search', __i18n_text_domain__ ) }
-						aria-hidden={ ! isOpenUnpinnedOrQueried }
-						className={ inputClass }
-						placeholder={ placeholder }
-						role="searchbox"
-						value={ searchValue }
-						ref={ this.searchInput }
-						onChange={ this.onChange }
-						onKeyUp={ this.keyUp }
-						onKeyDown={ this.keyDown }
-						onMouseUp={ this.props.onClick }
-						onFocus={ this.onFocus }
-						onBlur={ this.onBlur }
-						disabled={ this.props.disabled }
-						autoCapitalize="none"
-						dir={ this.props.dir }
-						maxLength={ this.props.maxLength }
-						minLength={ this.props.minLength }
-						{ ...autocorrectProps }
-					/>
-					{ this.renderStylingDiv() }
-				</form>
-				{ shouldRenderRightOpenIcon ? this.renderOpenIcon() : this.closeButton() }
-			</div>
-		);
-	}
+	const renderStylingDiv = React.useCallback( () => {
+		if ( typeof overlayStyling === 'function' ) {
+			return (
+				<div
+					className="search-component__text-overlay"
+					ref={ overlay as React.RefObject< HTMLDivElement > }
+				>
+					{ overlayStyling( keyword ) }
+				</div>
+			);
+		}
+		return null;
+	}, [ overlayStyling, keyword ] );
 
-	renderOpenIcon(): JSX.Element {
-		const enableOpenIcon = this.props.pinned && ! this.state.isOpen;
+	const renderOpenIcon = React.useCallback( (): JSX.Element => {
+		const enableOpenIcon = pinned && ! isOpen;
 
 		return (
 			<Button
 				className="search-component__icon-navigation"
-				ref={ this.openIcon }
-				onClick={ enableOpenIcon ? this.openSearch : this.focus }
+				ref={ openIcon as React.RefObject< HTMLButtonElement > }
+				onClick={ enableOpenIcon ? openSearch : focus }
 				tabIndex={ enableOpenIcon ? 0 : undefined }
-				onKeyDown={ enableOpenIcon ? this.openListener : undefined }
-				aria-controls={ 'search-component-' + this.instanceId }
-				aria-label={ this.props.__( 'Open Search', __i18n_text_domain__ ) }
+				onKeyDown={ enableOpenIcon ? openListener : undefined }
+				aria-controls={ 'search-component-' + instanceId }
+				aria-label={ __( 'Open Search', __i18n_text_domain__ ) }
 			>
-				{ ! this.props.hideOpenIcon && (
+				{ ! hideOpenIcon && (
 					// `className` is accepted for some reason the intrisic attributes for SVG won't allow it (but it does work)
 					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 					/* @ts-ignore */
```

Comment:
These comments can be all removed. They probably were true with some old types for `react` or `@wordpress/icons`, but today both `<Icon />` and `React.SVGProps` accept `className` without complaining.